src/payload.rs

```diff
@@ -163,7 +319,20 @@ where
     ///
     /// This can be used to modify the pipe, length, etc. of the
     /// packet.
-    pub fn update_header(&mut self, header: EsbHeader) {
+    ///
+    /// ## NOTE:
+    ///
+    /// The `length` of the packet can not be increased, only shrunk. If a larger
+    /// payload is needed, you must drop the current payload grant, and obtain a new
+    /// one. If the new header has a larger `length` than the current `length`, then
+    /// it will be truncated.
+    pub fn update_header(&mut self, mut header: EsbHeader) {
+        // TODO(AJM): Technically, we could drop the current grant, and request a larger one
+        // here, and it would totally work. However for now, let's just truncate
```

Comment:
```suggestion
        // here, and it would totally work. However for now, let's just truncate, because growing
        // the buffer would first have to be implemented in BBQueue.
```